tests/controllers/test_response_correlation_controller.py

```diff
@@ -114,3 +115,18 @@ def test_get_first_observation_x_invalid():
     df_observation = pd.DataFrame([int(1)], columns=["x_axis"])
     with pytest.raises(ValueError, match="invalid obs_data type"):
         _get_first_observation_x(df_observation)
+
+
+@pytest.mark.parametrize(
+    "raw_value,expected_formatted_value",
+    [
+        ("2022-08-05 14:25:00", "2022-08-05"),
+        ("2022-09-21 14:25:00", "2022-09-21"),
+        (14, "14"),
+        ("213", "213"),
+        ("SPAM", "SPAM"),
+    ],
+)
```

Comment:
Very nice, I would just suggest the order of the tests and rename one of them. In this way, the test cases are clearly expressing what is written in the string of the function:
```
@pytest.mark.parametrize(
    "raw_value,expected_formatted_value",
    [
        ("2022-08-05 14:25:00", "2022-08-05"),
        ("2022-09-21 14:25:00", "2022-09-21"),
        ("PARSING_FAILS_FROM_NOW", "PARSING_FAILS_FROM_NOW"),
        (14, "14"),
        ("213", "213"),
    ],
)
```

Reply:
very good suggestion - it seems like we can actually add descriptive names to our test cases, cf. [the pytest parametrize docs on test IDs](https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids)

i'm gonna try to have a go at that!

Reply:
I want to insist a little bit on keeping the `SPAM` as a silly string, as it seems to be [the pythonic way](https://docs.python.org/3.10/library/pprint.html) (see first code block) - but it's not a hard insist : )

Reply:
considering your suggestion of `PARSINGS_FAILS_FROM_NOW`, I would like to avoid adding that information ("parsing fails") to the test cases - i think it's an implementation details of the function. what the test wants to know, is: if I put in something that is not a date, will I get out the same thing, possibly converted to a string if it was a number.

does that sound OK? pushed the updated version with IDs

Reply:
That is fine!